addon/controllers/messages_base.js

```diff
@@ -35,11 +41,13 @@ export default Ember.Controller.extend({
 
   messages: Ember.computed("allMessages.[]", "offer", "item", function() {
     var messages = this.get("allMessages");
-    messages = this.get("isItemThread") ?
-      messages.filterBy("itemId", this.get("item.id")) :
-      messages
-          .filterBy("offerId", this.get("offer.id"))
-          .filterBy("item", null);
+    messages = this.get("isItemThread")
+      ? messages
+          .filterBy("messageableType", "Item")
+          .filterBy("messageableId", this.get("item.id"))
+      : messages
+          .filterBy("messageableType", "Offer")
+          .filterBy("messageableId", this.get("offer.id"));
     return messages.filter(m => {
       return Boolean(m.get("isPrivate")) === this.get("isPrivate");
     });
```

Comment:
```suggestion
  messages: Ember.computed("allMessages.[]", "offer", "item", function() {
     const type = this.get("isItemThread") ? "item" : "offer";
     return this.get("allMessages")
          .filterBy("messageableType", _.capitalize(type))
          .filterBy("messageableId", this.get(`${type}.id`))
          .filter(m => Boolean(m.get("isPrivate")) === this.get("isPrivate"));
```

Reply:
another option would be to create computed properties for `recordType` and `recordId`

All optional suggestions btw